.github/workflows/model_uploader.yml

```diff
@@ -0,0 +1,321 @@
+        uses: actions/download-artifact@v2
+        with:
+          name: upload
+          path: ./upload/
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: ${{ secrets.MODEL_UPLOADER_AWS_REGION }}
+          role-to-assume: ${{ secrets.MODEL_UPLOADER_ROLE }}
+          role-session-name: model-uploading
+      - name: Copy Files to the Bucket
+        id: copying_to_bucket
+        run: |
+          aws s3 sync ./upload/ s3://opensearch-exp/ml-models/huggingface/sentence-transformers/
+          echo "upload_time=$(TZ='America/Los_Angeles' date "+%Y-%m-%d %T")" >> $GITHUB_OUTPUT
+    outputs:
+      upload_time: ${{ steps.copying_to_bucket.outputs.upload_time }}
+   
+  # Step 6: Update MODEL_UPLOAD_HISTORY.md & supported_models.json
+  history-update:
+    needs: model-uploading
+    runs-on: 'ubuntu-latest'
+    permissions:
+      id-token: write
+      contents: write
+      pull-requests: write
+    env:
+      model_info: ${{ github.event.inputs.model_id }} (v.${{ github.event.inputs.model_version }})(${{ github.event.inputs.tracing_format }})
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      - name: Set Up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: Install Packages
+        run:
+          python -m pip install mdutils
+      - name: Update Model Upload History
+        run: |
+          python utils/model_uploader/update_models_upload_history_md.py \
+            ${{ github.event.inputs.model_id }} \
+            ${{ github.event.inputs.model_version }} \
+            ${{ github.event.inputs.tracing_format }} \
+            -ed ${{ github.event.inputs.embedding_dimension }} \
+            -pm ${{ github.event.inputs.pooling_mode }} \
+            -u ${{ github.actor }} -t "${{ needs.model-uploading.outputs.upload_time }}"
+      - name: Create PR Body
+        id: create_pr_body
+        run: |
+          embedding_dimension=${{ github.event.inputs.embedding_dimension }}
+          pooling_mode=${{ github.event.inputs.pooling_mode }}
+          
+          pr_body="
+          - [ ] This PR made commit to only these three files: MODEL_UPLOAD_HISTORY.md, supported_models.json, and CHANGELOG.md.
+          - [ ] CHANGELOG.md has been updated by the workflow or by you if the workflow fails to do so. 
+          - [ ] Merge conflicts have been resolved.
+          
+          ========= Workflow Details ==========
+          - Workflow Name: ${{ github.workflow }}
+          - Workflow Run ID: ${{ github.run_id }}
+          - Workflow Initiator: @${{ github.actor }}
+          
+          ========= Model Information =========
+          - Model ID: ${{ github.event.inputs.model_id }}
+          - Model Version: ${{ github.event.inputs.model_version }}
+          - Tracing Format: ${{ github.event.inputs.tracing_format }}
```

Comment:
How will this support "BOTH" case? As we discussed we need to have two separate objects if the tracing option is "Both"

Reply:
It will raise one PR, but there will be two objects in `support_models.json` and `MODEL_UPLOAD_HISTORY.md`.